### PR TITLE
feat(parsers): Add prefix-by parser

### PR DESCRIPTION
**Pull request by ChucKN0risK**

### CODE REVIEW

File: parsers/prefix-by/prefix-by.spec.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import seeds from '../../tests/seeds';
+import prefixBy from './prefix-by.parser';
+import { Token } from '../../types';
+import libs from '../global-libs';
+
+describe('prefix-by', () => {
```
[the remark this answers is not in the document]
Comment (ChucKN0risK) [reply parsers/prefix-by/prefix-by.spec.ts:6]: types option is undefined ✅

File: parsers/prefix-by/README.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+## Interface
+
+```ts
+interface parser {
+  name: 'prefix-by';
+  options?: {
+    key?: string;
+    prefix: string;
+    types?: Array<string>;
```
[the remark this answers is not in the document]
Comment (AllanMichay) [reply parsers/prefix-by/README.md:16]: Wouldn't it be better to name the key `applyOn` or `applyOnTokens`? I really like the idea

[the remark this answers is not in the document]
Comment (AMoreaux) [reply parsers/prefix-by/README.md:16]: So I prefer `applyOn `

[the remark this answers is not in the document]
Comment (AMoreaux) [reply parsers/prefix-by/README.md:16]: @ChucKN0risK please rename the option

File: parsers/sort-by/sort-by.spec.ts [hunk 3]
```diff
@@ -4,13 +4,13 @@ import { Token } from '../../types';
 import libs from '../global-libs';
 
 describe('sort-by', () => {
-  it('Get tokens - apply parsers', async () => {
+  it('Sorts tokens alphabetically by their name', async () => {
```
Comment (AMoreaux) [inline parsers/sort-by/sort-by.spec.ts:7]: Rebased the PR to remove this change that already done in another PR

File: parsers/prefix-by/prefix-by.parser.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+export type InputDataType = Array<Record<string, any>>;
+export type OutputDataType = InputDataType;
+export type OptionsType = {
+  key?: string;
+  prefix: string; // pattern
+  types?: Array<string>;
+};
+
+export default async function (
+  tokens: InputDataType,
```
Comment (AMoreaux) [inline parsers/prefix-by/prefix-by.parser.ts:13]: Do you want to allow prefixes with string as input to add a comment after `to-css-custom-properties` for example?

**Review by AMoreaux — CHANGES_REQUESTED** [review]

File: parsers/prefix-by/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ChucKN0risK) [reply parsers/prefix-by/README.md:16]: I don't mind doing this but we should also update the `suffix-by` parser as it also has the same `types` property. This will cause a breaking change though.

WDYT?

File: parsers/sort-by/sort-by.spec.ts (hunk 3 above)
Comment (ChucKN0risK) [reply parsers/sort-by/sort-by.spec.ts:7]: Done ✅

Comment (AMoreaux) [reply parsers/sort-by/sort-by.spec.ts:7]: This change shouldn't be in this PR since it concern another parser

File: parsers/prefix-by/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (AMoreaux) [reply parsers/prefix-by/README.md:16]: Parsers are isolated, and options are scoped. You can safely use `applyOn` here.

If you want to add the option applyOn on suffix-by. Feel free. It's another parser so another subject.

File: parsers/prefix-by/prefix-by.parser.ts (hunk 4 above)
Comment (AMoreaux) [reply parsers/prefix-by/prefix-by.parser.ts:13]: @ChucKN0risK  ?

**Review by AMoreaux — CHANGES_REQUESTED** [review]

File: parsers/prefix-by/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ChucKN0risK) [reply parsers/prefix-by/README.md]: Done for `prefix-by` ✅

File: parsers/sort-by/sort-by.spec.ts [hunk 5]
```diff
@@ -10,7 +10,7 @@ describe('sort-by', () => {
     expect(JSON.stringify(libs._.sortBy(seeds().tokens, ['name']))).toEqual(JSON.stringify(result));
     return;
   });
-  it('Get tokens - apply parsers - wihtout options', async () => {
+  it('Get tokens - apply parsers - without options', async () => {
```
Comment (AMoreaux) [inline parsers/sort-by/sort-by.spec.ts:13]: This change shouldn't be here. Please, move it to another PR.

File: parsers/prefix-by/prefix-by.parser.ts (hunk 4 above)
Comment (AMoreaux) [reply parsers/prefix-by/prefix-by.parser.ts:13]: So you don't want to allow prefixing a string like the output of `to-css-custom-properties`?

**Review by AMoreaux — CHANGES_REQUESTED** [review]

File: parsers/prefix-by/prefix-by.parser.ts (hunk 4 above)
Comment (ChucKN0risK) [reply parsers/prefix-by/prefix-by.parser.ts:13]: I want but I don't know how to do it.

File: parsers/sort-by/sort-by.spec.ts (hunk 5 above)
Comment (ChucKN0risK) [reply parsers/sort-by/sort-by.spec.ts]: Fixed ✅

**Review by AMoreaux — APPROVED** [review]
